### PR TITLE
Add frax3pool to mainnet minichef rewards.

### PR DESCRIPTION
**Pull request by mjdatadog**
Total SDL Reward Allocation = 2_000_000
Reward Duration = 2 month (8 weeks)
New frax3pool allocation point set to 68
LP Token deployed at: 0x0785aDDf5F7334aDB7ec40cD785EBF39bfD91520

Transaction from running mutisig script:
Transaction was Mined
---------------------
Tx Hash: 0x7d8202add2cf45b1906d9b809ffac0c3f818c0f23e1ea8eb219fab7c1c83fe51
From: 0x0AF91FA049A7e1894F480bFE5bBa20142C6c29a9
To: 0x3F8E527aF4e0c6e763e8f368AC679c44C45626aE
Value: 0
Function: Gnosis Safe.execTransaction
Block: 14854984
Gas Used: 323009 / 332213 (97.2%)

```
Events In This Transaction
--------------------------
├── MiniChefV2 (0x691ef79e40d909C715BE5e9e93738B3fF7D58534)
│   ├── LogUpdatePool
│   │   ├── pid: 1
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 3362567772049937137407
│   │   └── accSaddlePerShare: 784632825187762
│   ├── LogUpdatePool
│   │   ├── pid: 2
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 2714813703804314519604987
│   │   └── accSaddlePerShare: 693644573086
│   ├── LogUpdatePool
│   │   ├── pid: 3
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 3135163716979217455505100
│   │   └── accSaddlePerShare: 344030048809
│   ├── LogUpdatePool
│   │   ├── pid: 4
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 44829237755811034884
│   │   └── accSaddlePerShare: 17495933454597046
│   ├── LogUpdatePool
│   │   ├── pid: 5
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 139259787677052418057
│   │   └── accSaddlePerShare: 9490758208433711
│   ├── LogUpdatePool
│   │   ├── pid: 6
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 168779966501872592304581
│   │   └── accSaddlePerShare: 650563636730
│   ├── LogUpdatePool
│   │   ├── pid: 7
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 19617062321791123007718
│   │   └── accSaddlePerShare: 818260890657
│   ├── LogUpdatePool
│   │   ├── pid: 8
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 224441296635887591725
│   │   └── accSaddlePerShare: 262881613107842615
│   ├── LogUpdatePool
│   │   ├── pid: 9
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 1768460622105518364661242
│   │   └── accSaddlePerShare: 444013589113521
│   ├── LogUpdatePool
│   │   ├── pid: 10
│   │   ├── lastRewardTime: 1653666543
│   │   ├── lpSupply: 70364514331017481192263
│   │   └── accSaddlePerShare: 168833205178
│   ├── LogSaddlePerSecond
│   │   └── saddlePerSecond: 1635846757684555293
│   └── LogPoolAddition
│       ├── pid: 11
│       ├── allocPoint: 68
│       ├── lpToken: 0x0785aDDf5F7334aDB7ec40cD785EBF39bfD91520
│       └── rewarder: 0x0000000000000000000000000000000000000000
│
└── Gnosis Safe (0x3F8E527aF4e0c6e763e8f368AC679c44C45626aE)
    └── ExecutionSuccess
        ├── txHash: 0x13b88e6182d6d02d4ac95bc711caaa9aedac348e5fd8679453e260f340040686
        └── payment: 0
```

### CODE REVIEW

File: scripts/2022_05_26_add_frax3pool_to_minichef.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    """Add SimpleRewarder for T to minichef"""
+
+    print(f"You are using the '{network.show_active()}' network")
+    deployer = accounts.load("deployer")  # prompts for password
+    multisig = ApeSafe(MULTISIG_ADDRESSES[CHAIN_IDS["MAINNET"]])
+
+    minichef = multisig.contract(MINICHEF_ADDRESSES[CHAIN_IDS["MAINNET"]])
+    poolLength = minichef.poolLength()
+    newPoolIndex = poolLength + 1
+    minichef.massUpdatePools(["1", "2", "3", "4", "5", "6", "7", "8", "9", "10"])
```
Comment (hammeiam) [inline scripts/2022_05_26_add_frax3pool_to_minichef.py:18]: Could do something like `[str(x) for x in range(1, poolLength + 1)]`

File: scripts/2022_05_26_add_frax3pool_to_minichef.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+def main():
+    """Add SimpleRewarder for T to minichef"""
+
+    print(f"You are using the '{network.show_active()}' network")
+    deployer = accounts.load("deployer")  # prompts for password
+    multisig = ApeSafe(MULTISIG_ADDRESSES[CHAIN_IDS["MAINNET"]])
+
+    minichef = multisig.contract(MINICHEF_ADDRESSES[CHAIN_IDS["MAINNET"]])
+    poolLength = minichef.poolLength()
```
Comment (hammeiam) [inline scripts/2022_05_26_add_frax3pool_to_minichef.py:16]: prefer to use snake_case for pythonic variable names

File: scripts/2022_05_26_add_frax3pool_to_minichef.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    # Static
+    currentSDLPerSecond = minichef.saddlePerSecond()
+    frax3poolSDLPerSecond = (2_000_000 * 1e18) / (2 * 4 * 7 * 24 * 60 * 60)
+    totalAlloc = minichef.totalAllocPoint()
+
+    # Calculation
+    newSDLPerSecond = currentSDLPerSecond + frax3poolSDLPerSecond
+    fraxAllocationRate = frax3poolSDLPerSecond / newSDLPerSecond
+    fraxAlloc = round((totalAlloc * fraxAllocationRate) / (1 - fraxAllocationRate))
```
Comment (hammeiam) [inline scripts/2022_05_26_add_frax3pool_to_minichef.py:28]: I checked this math by confirming that `fraxAlloc / (totalAlloc + fraxAlloc) = fraxAllocationRate`

**Review by hammeiam — APPROVED** [review]
logic looks good to me

File: scripts/2022_05_26_add_frax3pool_to_minichef.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    print("fraxAlloc: ", fraxAlloc)
+    print("newPoolIndex: ", newPoolIndex)
+
+    # State Change
+    minichef.setSaddlePerSecond(newSDLPerSecond)
+    minichef.add(
+        newPoolIndex,
+        "0x0785aDDf5F7334aDB7ec40cD785EBF39bfD91520",
+        "0x0000000000000000000000000000000000000000",
+    )
```
Comment (penandlim) [inline scripts/2022_05_26_add_frax3pool_to_minichef.py:38]: The first parameter of the add call is the `allocpoint`, not the index. Index will be deterimned automatically by the contract.

https://github.com/saddle-finance/saddle-contract/blob/a5cc791b212420a85afcfd8472ef1179ef120841/contracts/rewards/MiniChefV2.sol#L108-L117

**Review by penandlim — APPROVED** [review]

File: scripts/2022_05_26_add_frax3pool_to_minichef.py (hunk 1 above)
Comment (hammeiam) [reply scripts/2022_05_26_add_frax3pool_to_minichef.py]: @mjdatadog fyi range is not inclusive of the last number, so range (1,5) will be 1,2,3,4

File: scripts/2022_05_26_add_frax3pool_to_minichef.py (hunk 3 above)
Comment (mjdatadog) [reply scripts/2022_05_26_add_frax3pool_to_minichef.py]: was changed to `pids = [x for x in range(1, pool_length)]`

File: scripts/2022_05_26_add_frax3pool_to_minichef.py (hunk 4 above)
Comment (mjdatadog) [reply scripts/2022_05_26_add_frax3pool_to_minichef.py]: fixed in https://github.com/saddle-finance/multisig-scripts/pull/12/commits/8a2f731c00f3e397bd123fe2f6dd582509868243

**Pull request merged**